### ISSUE TITLE
reading the response is taking too long - 10 minutes

### ISSUE THREAD

**Issue by itamarzil123** [sfackler/rust-native-tls#286]
I am using native_tls::{TlsConnector, TlsStream} as part of building a browser project in Rust.
`reader.read_to_end(...) -` this read operation hangs for about 10 minutes - and only after this enormous delay, the html contents of the URL (www.bbc.com for example) is resolved. 
for comparison - in browsers like chrome - takes 2 seconds for www.bbc.com, in curl takes 2-5 seconds.
any thoughts on why this process my take so long ? (nothing serious on the wireshark logs, handshake times seems reasonable)
simplified version of my code:

```
let mut response = Vec::new();
let mut tcp_stream = TcpStream::connect((ip, protocol_port)).expect("Error...");
let connector = TlsConnector::new().expect("Error creating TLS connector");
let mut tls_stream = connector.connect(host.as_str(), tcp_stream).expect("Error connecting with TLS");
tls_stream.write_all(request.as_bytes()).expect("Error....");
let mut reader = io::BufReader::new(tls_stream);
let mut buffer = [0; 4096]; 
reader.read_to_end(&mut response).expect("error...");
note: tried with a buffer, tried with 'read' instead of read_to_end
```

**Comment by sfackler** [comment]
read_to_end reads until the connection closes. You need to parse the response to determine when it is complete.

**Comment by itamarzil123** [comment]
@sfackler thank you for your answer. 
as I mentioned the code I shosed is a simplified part. the rest of the code includes:
 let response_str = String::from_utf8_lossy(&response);
println!("response_str: {}", response_str);
and also writing to files.
the operation is successfull, however it takes 10 minutes to complete (unlike using curl / browser which takes 2 seconds)
I tried buffering and didn't work.
any thoughts ? or even how to debug ?
